tests/test_sessions.py

```diff
@@ -0,0 +1,209 @@
+
+        # cookie = get_cookie_from_client(client)
+        # pdb.set_trace()
+        assert len(client_two.cookie_jar) == 0
+        client_two.cookie_jar = client.cookie_jar
+        response = client.get(flask_security.url_for_security('logout'))
+
+    with client_two as client:
+        assert not flask.session
+        # pdb.set_trace()
+        assert not testutils.client_authenticated(client)
+
+
+def test_multiple_clients_single_user(app):
+    """Test session generation when logging in as the same user with multiple
+    clients."""
+    InvenioAccounts(app)
+    app.register_blueprint(blueprint)
+
+    user = testutils.create_test_user()
+
+    clients_no = 2
+    clients = [app.test_client() for i in range(0, clients_no)]
+
+    for client in clients:
+        testutils.login_user_via_view(client, user.email,
+                                      user.password_plaintext)
+    assert len(app.kvsession_store.keys()) == clients_no
+
+    # sid_s/kvsession_store keys should be unique
+    for key in app.kvsession_store.keys()[1:clients_no]:
+        assert app.kvsession_store.keys()[0] != key
+
+
+def test_logout_listener(app):
+    """Test sessions.py:logout_listener"""
+    InvenioAccounts(app)
+    app.register_blueprint(blueprint)
+
+    user_logged_out.connect(logout_listener, app)
+    user = testutils.create_test_user()
+    user_2 = testutils.create_test_user(email="testz@ss.org")
+
+    with app.test_client() as client:
+        testutils.login_user_via_view(client, user.email,
+                                      user.password_plaintext)
+        assert len(app.kvsession_store.keys()) == 1
+
+        kvsession_key = app.kvsession_store.keys()[0]
+        sesh = app.kvsession_store.get(kvsession_key)
+        cookie = get_cookie_from_client(client, kvsession_key)
+        cookie_sid_s = get_sid_s_from_cookie(cookie)
+        assert kvsession_key == cookie_sid_s
+
+        import pdb
+        # Logout by posting to url_for_security('logout')?
+        client.get(flask_security.url_for_security('logout'))
+        assert not testutils.client_authenticated(client)
+        assert len(app.kvsession_store.keys()) == 1
+        # Is the session removed from the kvsession store?
+        # if the user logs in again, is the second session different
+        # from the first?
+        # pdb.set_trace()
+        testutils.login_user_via_view(client, user_2.email,
+                                      user_2.password_plaintext)
+        assert len(app.kvsession_store.keys()) == 1
+        assert app.kvsession_store.keys()[0] == kvsession_key
+        assert True
+
+
+def get_cookie_from_client(client, sid_s=None):
+    """Extracts the cookie from the client's cookie jar matching the given
+    sid_s value. If sid_s is none, it returns the first cookie."""
+    for c in client.cookie_jar:
+        if sid_s is None or sid_s in c.value:
+            return c
+
+
+def get_sid_s_from_cookie(cookie):
```

Comment:
`testutils.py` is more befitting it: It complements `get_cookie_from_client`, which should only work for instances of Flask's `test_client`.